operator/k8s_cep_gc.go

```diff
@@ -64,34 +81,61 @@ func enableCiliumEndpointSyncGC() {
 							Errorf("Saw %T object while expecting *cilium_v2.CiliumEndpoint", cepObj)
 						continue
 					}
-
 					cepFullName := cep.Namespace + "/" + cep.Name
-					_, exists, err := watchers.PodStore.GetByKey(cepFullName)
-					if err != nil {
-						scopedLog.WithError(err).Warn("Unable to get pod from store")
-						continue
-					}
-					if !exists {
-						// FIXME: this is fragile as we might have received the
-						// CEP notification first but not the pod notification
-						// so we need to have a similar mechanism that we have
-						// for the keep alive of security identities.
-						scopedLog = scopedLog.WithFields(logrus.Fields{
-							logfields.EndpointID: cep.Status.ID,
-							logfields.K8sPodName: cepFullName,
-						})
-						scopedLog.Debug("Orphaned CiliumEndpoint is being garbage collected")
-						PropagationPolicy := meta_v1.DeletePropagationBackground // because these are const strings but the API wants pointers
-						err := ciliumClient.CiliumEndpoints(cep.Namespace).Delete(
-							ctx,
-							cep.Name,
-							meta_v1.DeleteOptions{PropagationPolicy: &PropagationPolicy})
+					scopedLog = scopedLog.WithFields(logrus.Fields{
+						logfields.K8sPodName: cepFullName,
+					})
+
+					if !once {
+						// If we are running this function "once" it means that we
+						// will delete all CEPs in the cluster regardless of the pod
+						// state therefore we won't even watch for the pod store.
```

Comment:
This comment doesn't belong inside the "if" because it mentions the case where once is true. It can be confusing to read.

```suggestion
					// If we are running this function "once" it means that we
					// will delete all CEPs in the cluster regardless of the pod
					// state therefore we won't even watch for the pod store.
					if !once {
```